demo

```diff
@@ -211,45 +233,72 @@ done
 
 
 echo "Creating a new user..."
-UPLOAD_ARTIFACT="true"
-RETRIES=0
-while :
-do
+UPLOAD_ARTIFACT=1
+PRINT_LOGIN_INFO=0
+PRINT_USER_EXISTS=1
+if [ $ENTERPRISE = 1 ]; then
+    TENANT_ID=$(docker exec \
+            ${DOCKER_COMPOSE_PROJECT_NAME}_mender-tenantadm_1 \
+            /usr/bin/tenantadm create-org \
+            --name=DemoOrganization \
+            --username=${USER} \
+            --password=${PASSWORD})
+    retval=$?
+    EXISTS_ERROR=1
+    if [ $retval -eq 0 ]; then
+        TENANT_ID=$(echo "$TENANT_ID" | tr -d '\r')
+    else
+        TENANT_ID=
+    fi
+else
     docker exec \
-             ${DOCKER_COMPOSE_PROJECT_NAME}_mender-useradm_1 \
-             /usr/bin/useradm create-user \
-             --username=${USER} \
-             --password=${PASSWORD} \
-             > /dev/null
+            ${DOCKER_COMPOSE_PROJECT_NAME}_mender-useradm_1 \
+            /usr/bin/useradm create-user \
+            --username=${USER} \
+            --password=${PASSWORD} \
+            > /dev/null
     retval=$?
-    case $retval in
-        0)
-            echo "****************************************"
-            echo
-            echo "Username: mender-demo@example.com"
-            echo  "Login password: ${PASSWORD}"
-            echo
-            echo "****************************************"
-            echo "Please keep the password available, it will not be cached by the login script."
-            break ;;
-        5) # If the user exists, skip uploading the Artifact
-            UPLOAD_ARTIFACT=""
-            echo "The user already exists. Skipping"
-            echo "If you don't remember the password, you can run '$(basename $0) down' to delete"
-            echo "the old user and rerun '$(basename $0) up' to create a new one."
-            echo "Please note that all data will be deleted from the old demo server."
-            break ;;
-        *) echo "docker exec error: " $retval; exit $retval ;;
-    esac
-    if [[ $RETRIES -ge $RETRY_LIMIT ]]; then
-        echo "Retried $RETRIES times without success. Giving up."
-        exit 1
+    EXISTS_ERROR=5
```

Comment:
So the retcode for existing user is different from Enterprise to OS?

Reply:
Yes, I actually started by trying to unify the two binaries' return codes, but there are TONS of different error codes in both of them, 5 is already used for something else in tenantadm, and the operations don't map well to each other. In the end I gave it up, and took it into account here instead.